app/src/main/java/space/taran/arknavigator/ui/fragments/SettingsFragment.kt

```diff
@@ -0,0 +1,149 @@
+        savedInstanceState: Bundle?
+    ): View {
+        super.onCreateView(inflater, container, savedInstanceState)
+
+        Log.d(SETTINGS_SCREEN, "inflating layout for SettingsFragment")
+        binding = FragmentSettingsBinding.inflate(inflater, container, false)
+        presenter.onCreateView()
+
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        Log.d(SETTINGS_SCREEN, "view created in ResourcesFragment")
+        super.onViewCreated(view, savedInstanceState)
+
+        App.instance.appComponent.inject(this)
+
+        binding.apply {
+            crashReportSwitch.setOnCheckedChangeListener { compoundButton, b ->
+                presenter.onCrashReportingClick(b)
+            }
+
+            cacheReplicationSwitch.setOnCheckedChangeListener { compoundButton, b ->
+                presenter.onImgCacheReplicationClick(b)
+            }
+
+            indexReplicationSwitch.setOnCheckedChangeListener { compoundButton, b ->
+                presenter.onIndexReplicationClick(b)
+            }
+
+            crashInfo.setOnClickListener {
+                showInfoDialog(R.string.what_are_crash_reports_, R.string.crash_reports_explanation)
+            }
+
+            imgCacheInfo.setOnClickListener {
+                showInfoDialog(R.string.what_is_image_replication_, R.string.explanation_of_this_feature)
+            }
+
+            indexReplicationInfo.setOnClickListener {
+                showInfoDialog(R.string.what_is_index_replication_, R.string.explanation_of_this_feature)
+            }
+
+            resetPreferences.setOnClickListener {
+                val dialog = ConfirmationDialogFragment.newInstance(
+                    getString(R.string.are_you_sure_),
+                    getString(R.string.all_preferences_will_be_reset_to_default),
+                    getString(R.string.yes),
+                    getString(R.string.no)
+                )
+                dialog.show(childFragmentManager, ConfirmationDialogFragment.CONFIRMATION_DIALOG_TAG)
+            }
+        }
+    }
+
+    override fun init() {
+        Log.d(SETTINGS_SCREEN, "initializing SettingsFragment")
+        (activity as MainActivity).setSelectedTab(0)
+        (activity as MainActivity).setToolbarVisibility(false)
+        (requireActivity() as MainActivity).setBottomNavigationVisibility(true)
+
+        childFragmentManager.setFragmentResultListener(
+            ConfirmationDialogFragment.POSITIVE_KEY,
+            this
+        ) { _, _ ->
+            presenter.onResetPreferencesClick()
+        }
+    }
+
+    private fun showInfoDialog(
+        @StringRes titleRes: Int? = null,
+        @StringRes descriptionRes: Int? = null,
+        titleString: String? = null,
+        descriptionString: String? = null
+    ) {
+        val title = titleString ?: titleRes?.let { getString(it) }
+        ?: throw AssertionError("No value passed for title!")
```

Comment:
Good that you want to verify invariants, but here it looks a bit strange.
`titleString` is loaded earlier, so we can verify it there and make the argument non-nullable.
But we control state of these String variables, so it's redundant here.

`AssertionError` is useful when we have implicit invariants, some logical conditions we assume to be true.
Here the variable comes just from our condition, so it is explicit.

Also, it's not really critical to verify here, because user will not loose her data.

Reply:
Btw, maybe some of `AssertionError`s placed by me should be normal exceptions in fact. We should check.